### PR TITLE
Add implementation for grallochandler Map, Unmap functions

### PR DESCRIPTION
**Pull request by JeevakaPrabu**
Change-Id: Ic2377b02413f7536784cd2560e16bedfc4426992
Tracked-On: None
Signed-off-by: Jeevaka Prabu Badrappan <jeevaka.badrappan@intel.com>

### CODE REVIEW

**Comment by JeevakaPrabu** [comment]
@kalyankondapally @harishkrupo

File: os/android/gralloc1bufferhandler.cpp [hunk 1]
```diff
@@ -129,15 +133,43 @@ void Gralloc1BufferHandler::CopyHandle(HWCNativeHandle source,
...
+  rect.top = y;
+  rect.width = width;
+  rect.height = height;
+
+  gralloc1_device_t *gralloc1_dvc =
+      reinterpret_cast<gralloc1_device_t *>(device_);
+  uint32_t status = lock_(gralloc1_dvc, handle->imported_handle_, GRALLOC_USAGE_SW_READ_RARELY,
+          GRALLOC_USAGE_SW_READ_RARELY, &rect, map_data, acquireFence);
+  ALOGI("%s - lock returned: %u", __FUNCTION__, status);
   return NULL;
```
Comment (kalyankondapally) [inline os/android/gralloc1bufferhandler.cpp]: retrun NULL here is confusing, probably we should share the API ?

**Comment by kalyankondapally** [comment]
@yugang @js0701 @Kishore409 @harishkrupo  please review

File: os/android/gralloc1bufferhandler.cpp (hunk 1 above)
Comment (harishkrupo) [reply os/android/gralloc1bufferhandler.cpp]: yeah, we can just return the status instead.

File: os/android/gralloc1bufferhandler.cpp [hunk 2]
```diff
@@ -129,15 +133,43 @@ void Gralloc1BufferHandler::CopyHandle(HWCNativeHandle source,
...
+  auto gr_handle = (struct cros_gralloc_handle *)handle->imported_handle_;
+  if (!gr_handle) {
+    ETRACE("could not find gralloc drm handle");
+    return;
+  }
+
+  int releaseFence = 0;
+  gralloc1_device_t *gralloc1_dvc =
+      reinterpret_cast<gralloc1_device_t *>(device_);
+  unlock_(gralloc1_dvc, handle->imported_handle_, &releaseFence);
```
Comment (harishkrupo) [inline os/android/gralloc1bufferhandler.cpp]: Similarly we can return the status here.

File: os/android/gralloc1bufferhandler.cpp (hunk 1 above)
Comment (js0701) [reply os/android/gralloc1bufferhandler.cpp]: I think the map api is to return the vritual interface CPU can access? That is at leat the story in Linux side. Linux is directly this API to gbm_bo_map

File: os/android/gralloc1bufferhandler.cpp (hunk 2 above)
Comment (js0701) [reply os/android/gralloc1bufferhandler.cpp]: this API is inheriting from NativeBufferHandler. gbm_bo_umap is a void function too

**Comment by kalyankondapally** [comment]
@JeevakaPrabu Are you going to update the full feature of dumping the buffers being sent to display ?

**Comment by JeevakaPrabu** [comment]
@kalyankondapally : Dumping the buffers will be delivered as a separate patch. New pull request for this changes is https://github.com/intel/IA-Hardware-Composer/pull/132

**Pull request closed without merging**